flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/merging/FileMergingCheckpointUtils.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.merging;
+
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+
+import javax.annotation.Nullable;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/** Utilities for file-merging checkpoints. */
+public class FileMergingCheckpointUtils {
+
+    // ------------------------------------------------------------------------
+    //  SSM initialization
+    // ------------------------------------------------------------------------
+
+    /** A class that packs the file system info for snapshot. */
+    public static class SnapshotFileSystemInfo {
+        FileSystem fs;
+        Path checkpointBaseDirectory;
+        Path sharedStateDirectory;
+        Path taskOwnedStateDirectory;
+        @Nullable Path defaultSavepointDirectory;
+
+        public SnapshotFileSystemInfo(
+                FileSystem fs,
+                Path checkpointBaseDirectory,
+                @Nullable Path sharedStateDirectory,
+                Path taskOwnedStateDirectory,
+                @Nullable Path defaultSavepointDirectory) {
+            this.fs = fs;
+            this.checkpointBaseDirectory = checkpointBaseDirectory;
+            this.sharedStateDirectory = sharedStateDirectory;
+            this.taskOwnedStateDirectory = taskOwnedStateDirectory;
+            this.defaultSavepointDirectory = sharedStateDirectory;
```

Comment:
=defaultSavepointDirectory？

Reply:
I removed the savepoint directory since it will not be used in the near future